### PR TITLE
[Infineon] Integrate OTA to Existing Infineon P6 examples

### PR DESCRIPTION
**Pull request by praveenCY**
#### Problem
Infineon P6 examples are not integrated with Software image update support

#### Change overview

- Integrated OTA with all P6 example apps
- Added support for parse OTA header in OTAImageProcessorImpl.cpp
- Added Software image update documentation for Infineon P6
- Added base and update build scripts for  all example apps
- Moved Bootloader and OTA related files to third_party/p6 folder instead of ota_requestor app

#### Testing
Manually Verified OTA Update for all Infineon P6 examples

### CODE REVIEW

File: .github/.wordlist.txt [hunk 1]
```diff
@@ -312,6 +312,7 @@ customizations
 cvfJ
 cxx
 CY
+cypressprogrammer
```
Comment (andy31415) [inline .github/.wordlist.txt:315]: cypresprogrammer is not a word.

Use URL tags or backticks to not need to add these as words

File: .github/.wordlist.txt [hunk 2]
```diff
@@ -618,6 +619,7 @@ idx
 ifconfig
 ifdef
 ifdefs
+ifx
```
Comment (andy31415) [inline .github/.wordlist.txt:622]: similar here and below. ifx and softwaretools are not words

**Review by woody-apple — APPROVED** [review]
Fast tracking platform changes

File: .github/.wordlist.txt (hunk 2 above)
Comment (woody-apple) [reply .github/.wordlist.txt:622]: we have lots of words (even listed above that aren't words)

File: .github/.wordlist.txt (hunk 1 above)
Comment (woody-apple) [reply .github/.wordlist.txt:315]: we have lots of words (even listed above that aren't words)

Comment (praveenCY) [reply .github/.wordlist.txt]: @andy31415 : Remove the words by adding URL tags

**Review by andy31415 — APPROVED** [review]
Approved, offline discussion with @praveenCY that we will move the build steps into a variant of build_examples (so that the ota_*build.sh scripts are not needed since having various shell scripts makes it harder to maintain the sdk in a generic manner).

**Pull request merged**